### ISSUE TITLE
Enhancement: Searching through records - file structure visibility?

### ISSUE THREAD

**Issue by vns2** [janastu/ncbs-omeka-backbone-client#95]
Based on user feedback.

Need to find a way to search through data. Say, finding all audio files by Person X. This could be a simple display of the Omeka database. But need to discuss how and where to display such info.

**Comment by salus-sage** [comment]
@vns2  what is file structure visibility?

**Comment by vns2** [comment]
@salus-sage This issue is basically about having search functionality all across the site. One needs to search through the file names and annotations in the first version of 'search'. By file structure visibility, I meant: one way would be to show up the file detail from Omeka database onto the exhibit. Almost like an index page of all files. And then, when you click on one of the audio or video or image files, you go to that particular page in the exhibit. But this is not very elegant. We'd have to think through this a little more. I think the better steps would be:
-- Having a permanent URL for omeka files that can be located inside the exhibit pages.
-- Being able to search through file names and annotations/tags.
-- Typing in something in a search box results in a drop down of all available files with that word.
-- Clicking on a file results in going to the exhibit page where that file is located. A sub-theme or space tour or peopleplex, etc.
-- In peopleplex, one could have a dropdown list of all nodes so that people can directly go to a particular person's node.

**Comment by salus-sage** [comment]
i think that is to complicate the search feature. 

- the search feature may be to search the "Archive" for items and view it's details individually in "search results", while showing which collection it belongs to and/or search for the exhibits itself. 

- Going back to the exhibit at the exact resource -> doesn't sound like a search feature to me, it will be great if you can point me to any examples online. 

- And about routing to Sandbox - let me confirm again that sandbox is not coming from omeka and is not indexed. 

What is the direction we need to take further with sandbox is still unclear to me. 
In any case the entire scope of this issue includes lot of other  issues, not related to just search feature.

**Comment by vns2** [comment]
First, one thing to just keep in mind since this will be critical in the future. All the content we have on the site right now is for an 'exhibit', not an 'archive'. There is no archive yet. The 'archive' content will be much larger and it will be like going into the catalog of a library. The exhibit right now is a curation of material from this original set of material (you have not seen that; it is on my desktop). What we are going toward, ideally in the next six months, is a front page for an archive with three tabs: (1) Browse the archive (2) Enter the exhibit -- what we have right now (3) Search -- this will be an evolving function, as we figure out what to search and how to display the result.

I think we can mark this issue as a March 2017 delivery item. But this is the way I'd like to define the scope of this issue, and see if it works for you:

Use case: User is in exhibit main page. They have a search box, say next to top three tabs - NCBS/About/Sandbox. They write "Sowdhamini" in search box. That search is through (a) all written material in theme pages (b) all metadata in theme page content (c) all metadata in Sandbox content. 
This leads to a search result page of location within exhibit.  Links that lead directly to, eg:
-- Photo 7 of slideshow 1 in Chapter 1 in Theme 4 
-- Node X in Peopleplex
-- Written part in Chapter 2 in Theme 3
-- Room X, Audio Y in SpaceTour

**Comment by salus-sage** [comment]
So, this implies that a pre-condition for the search feature is to re-visit
Sandbox
and it's contents be updated with taxonomy/tags and archived in omeka too!

**Comment by vns2** [comment]
Yes. In the long term, we should always think of the 'archive' as being all content, all stored in one accessible location. From there, one uses the taxonomy/tags to help build an exhibit. This would be the clean, scaleable approach, I think?

**Comment by salus-sage** [comment]
Now, that makes sense.
i'll plan milestones accordingly and share with you when ready.
